g2core/settings/settings_shapeoko2.h

```diff
@@ -156,70 +157,91 @@
+#define DI1_ENABLED                 IO_ENABLED
+#define DI1_POLARITY                NORMALLY_CLOSED
 //#define DI1_ACTION                  INPUT_ACTION_STOP
-#define DI1_ACTION                  INPUT_ACTION_NONE
-#define DI1_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI1_ACTION                  INPUT_ACTION_LIMIT
 
 // Xmax
-#define DI2_MODE                    NORMALLY_CLOSED
+#define DI2_ENABLED                 IO_ENABLED
+#define DI2_POLARITY                NORMALLY_CLOSED
 //#define DI2_ACTION                  INPUT_ACTION_STOP
-#define DI2_ACTION                  INPUT_ACTION_NONE
-#define DI2_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI2_ACTION                  INPUT_ACTION_LIMIT
 
 // Ymin
-#define DI3_MODE                    NORMALLY_CLOSED
+#define DI3_ENABLED                 IO_ENABLED
+#define DI3_POLARITY                NORMALLY_CLOSED
 //#define DI3_ACTION                  INPUT_ACTION_STOP
-#define DI3_ACTION                  INPUT_ACTION_NONE
-#define DI3_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI3_ACTION                  INPUT_ACTION_LIMIT
 
 // Ymax
-#define DI4_MODE                    NORMALLY_CLOSED
+#define DI4_ENABLED                 IO_ENABLED
+#define DI4_POLARITY                NORMALLY_CLOSED
 //#define DI4_ACTION                  INPUT_ACTION_STOP
-#define DI4_ACTION                  INPUT_ACTION_NONE
-#define DI4_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI4_ACTION                  INPUT_ACTION_LIMIT
 
 // Zmin
-#define DI5_MODE                    IO_ACTIVE_HIGH   // Z probe
+#define DI5_ENABLED                 IO_ENABLED
+#define DI5_POLARITY                IO_ACTIVE_HIGH   // Z probe
 #define DI5_ACTION                  INPUT_ACTION_NONE
-#define DI5_FUNCTION                INPUT_FUNCTION_NONE
 
 // Zmax
-#define DI6_MODE                    NORMALLY_CLOSED
+#define DI6_ENABLED                 IO_ENABLED
+#define DI6_POLARITY                NORMALLY_CLOSED
 //#define DI6_ACTION                  INPUT_ACTION_STOP
-#define DI6_ACTION                  INPUT_ACTION_NONE
-#define DI6_FUNCTION                INPUT_FUNCTION_LIMIT
+#define DI6_ACTION                  INPUT_ACTION_LIMIT
 
 // Amin
-#define DI7_MODE                    IO_MODE_DISABLED
+#define DI7_ENABLED                 IO_DISABLED
+#define DI7_POLARITY                IO_ACTIVE_LOW
 #define DI7_ACTION                  INPUT_ACTION_NONE
-#define DI7_FUNCTION                INPUT_FUNCTION_NONE
 
 // Amax
-#define DI8_MODE                    IO_MODE_DISABLED
+#define DI8_ENABLED                 IO_DISABLED
+#define DI8_POLARITY                IO_ACTIVE_LOW
 #define DI8_ACTION                  INPUT_ACTION_NONE
-#define DI8_FUNCTION                INPUT_FUNCTION_NONE
 
 // Hardware interlock input
-#define DI9_MODE                    IO_MODE_DISABLED
+#define DI9_ENABLED                 IO_DISABLED
+#define DI9_POLARITY                IO_ACTIVE_LOW
 #define DI9_ACTION                  INPUT_ACTION_NONE
```

Comment:
And we should add input 10 back in:
```suggestion
#define DI9_ENABLED                 IO_DISABLED
#define DI9_POLARITY                IO_ACTIVE_LOW
#define DI9_ACTION                  INPUT_ACTION_NONE

#define DI10_ENABLED                 IO_DISABLED
#define DI10_POLARITY                IO_ACTIVE_LOW
#define DI10_ACTION                  INPUT_ACTION_NONE

```